thunder/torch/__init__.py

```diff
@@ -1589,6 +1589,37 @@ def where(
     return clang.where(pred, a, b)
 
 
+@torchsymbol(torch.nan_to_num, is_method=True)
+def nan_to_num(
+    a: TensorLike,
+    nan: None | Number = 0.0,
+    posinf: None | Number = None,
+    neginf: None | Number = None,
+    /,
+    out: None | TensorLike = None,
+):
+
+    utils.check(out is None, lambda: "out is not None which is currently unsupported", NotImplementedError)
+
+    if dtypes.is_boolean_dtype(a.dtype) or dtypes.is_integer_dtype(a.dtype):
+        # NOTE PyTorch returns a.clone()
+        return a
```

Comment:
To make it a clone, what about returning `a - 0` for integer types and `a | a` for boolean types?

Reply:
I can definitely do that!